nixarr/autobrr/default.nix

```diff
@@ -0,0 +1,202 @@
+  group = "autobrr";
+
+  # Define config format and template
+  configFormat = pkgs.formats.toml {};
+  configTemplate = configFormat.generate "autobrr.toml" cfg.settings;
+in {
+  options.nixarr.autobrr = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      example = true;
+      description = ''
+        Whether or not to enable the Autobrr service.
+
+        **Required options:** [`nixarr.enable`](#nixarr.enable)
+      '';
+    };
+
+    package = mkPackageOption pkgs "autobrr" {};
+
+    openFirewall = mkOption {
+      type = types.bool;
+      defaultText = literalExpression ''!nixarr.autobrr.vpn.enable'';
+      default = !cfg.vpn.enable;
+      example = true;
+      description = "Open firewall for the Autobrr port.";
+    };
+
+    vpn.enable = mkOption {
+      type = types.bool;
+      default = false;
+      example = true;
+      description = ''
+        **Required options:** [`nixarr.vpn.enable`](#nixarr.vpn.enable)
+
+        Route Autobrr traffic through the VPN.
+      '';
+    };
+
+    settings = lib.mkOption {
+      type = lib.types.submodule {freeformType = configFormat.type;};
+      default = {
+        host = "0.0.0.0";
+        port = 7474;
+        checkForUpdates = false;
+      };
+      example = {
+        logLevel = "DEBUG";
+      };
+      description = ''
+        Autobrr configuration options.
+
+        See https://autobrr.com/configuration/autobrr for more information.
+
+        `sessionSecret` is automatically generated upon first installation and will be overridden.
+      '';
+    };
+
+    stateDir = mkOption {
+      type = types.path;
+      default = "${nixarr.stateDir}/autobrr";
+      defaultText = literalExpression ''"''${nixarr.stateDir}/autobrr"'';
+      example = "/nixarr/.state/autobrr";
+      description = "The location of the state directory for the Autobrr service.";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = cfg.vpn.enable -> nixarr.vpn.enable;
+        message = ''
+          The nixarr.autobrr.vpn.enable option requires the
+          nixarr.vpn.enable option to be set, but it was not.
+        '';
+      }
+      {
+        assertion = cfg.enable -> nixarr.enable;
+        message = ''
+          The nixarr.autobrr.enable option requires the nixarr.enable
+          option to be set, but it was not.
+        '';
+      }
+    ];
+
+    users = {
+      groups.${group} = {};
+      users.${user} = {
+        isSystemUser = true;
+        group = group;
+      };
+    };
+
+    # Create state directory with proper permissions
+    systemd.tmpfiles.rules = [
+      "d '${cfg.stateDir}' 0700 ${user} ${group} - -"
+    ];
+
+    # Configure the autobrr service
+    services.autobrr = {
+      enable = true;
+      package = cfg.package;
+      # We need to provide a secretFile even though we're handling it ourselves
+      secretFile = "/dev/null"; # This is a placeholder that won't be used
```

Comment:
From reading this, it's not immediately obvious how secrets are handled in the module, why is this set to null. I think it could be documented better in the options.

Otherwise this looks great, assuming that this is reasonably tested, we should just merge this.

Reply:
Yea, it's seems a little weird, I'll investigate more. Also, maybe we should add both options again (like for #44).

Reply:
Checked it again. So this option is not optional, and I did not find a way to "disable" it. The dummy value is needed. I'll enhance the docs.

Reply:
No this is fine.